packages/boxel-ui/addon/src/components/resizable-panel-group/index.gts

```diff
@@ -365,109 +408,50 @@ export default class ResizablePanelGroup extends Component<Signature> {
       this.panelGroupElement[this.perpendicularLengthProperty] <
       this.minimumLengthToShowHandles;
 
-    let panelLengths: number[] = [];
+    let panelLengths: number[] = Array.from(this.listPanelContext.values()).map(
+      (panelContext) => panelContext.lengthPx,
+    );
+    let newContainerSize = this.panelGroupLengthWithoutResizeHandlerPx;
+    if (newContainerSize == undefined) {
+      console.warn('Expected newContainerSize to be defined');
+      return;
+    }
+
     for (let index = 1; index <= this.listPanelContext.size; index++) {
-      let panelEl = this.panelGroupElement.querySelector(
-        `[id='${index}'].boxel-panel.${this.args.orientation}`,
-      ) as HTMLElement;
-      if (!panelEl) {
-        console.error(
-          `Could not find selector: [id='${index}'].boxel-panel when handling window resize for resizeable panel group`,
-        );
-        continue;
+      let panelContext = this.listPanelContext.get(index);
+      let panelRatio = this.panelRatio[index - 1];
+      if (!panelRatio) {
+        console.warn('Expected panelRatio to be defined');
+        break;
       }
-      panelLengths.push(panelEl[this.offsetLengthProperty]);
-    }
-    let totalPanelLength = sumArray(panelLengths);
-    let resizeHandlerSelector = `#resize-handler-${this.args.orientation}-1`;
-    let resizeHandlerEl = this.panelGroupElement.querySelector(
-      resizeHandlerSelector,
-    ) as HTMLElement;
-    if (!resizeHandlerEl) {
-      console.error(
-        `Could not find selector: ${resizeHandlerSelector} when handling window resize for resizeable panel group`,
+      let proportionalSize = panelRatio * newContainerSize;
+      let actualSize = Math.round(
+        panelContext?.minLengthPx
+          ? Math.max(proportionalSize, panelContext.minLengthPx)
+          : proportionalSize,
       );
-      return;
+      panelLengths[index - 1] = actualSize;
     }
 
-    let resizeHandleContainer = (resizeHandlerEl as HTMLElement).parentElement!;
-    let resizeHandlerLength = resizeHandleContainer[this.offsetLengthProperty];
-
-    let totalResizeHandlerLength =
-      resizeHandlerLength * (this.listPanelContext.size - 1);
-
-    let panelGroupWithoutResizeHandlerLength =
-      entry.contentRect[this.contentRectLengthProperty] -
-      totalResizeHandlerLength;
-    let panelGroupRemainingLength =
-      panelGroupWithoutResizeHandlerLength - totalPanelLength;
-
-    if (panelGroupRemainingLength === 0) {
-      // no resizing needed
-      return;
+    let remainingContainerSize = newContainerSize - sumArray(panelLengths);
+    let panelWithNoMinLength = Array.from(this.listPanelContext).find(
+      (panelContext) => panelContext[1].minLengthPx == undefined,
```

Comment:
panelContext name is confusing because this is a tuple with an integer and a PanelContext, not a PanelContext